src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -28,14 +30,47 @@
      * 26.04.2019 Сергей 7 100
      * 26.04.2019 София 9 100
      * 26.04.2019 Сергей 11 50
-     *
+     * <p>
      * Пример вывода:
      * Отчёт за период 01.04.2019  - 30.04.2019
      * Сергей - 1550
      * Андрей - 600
      * София - 900
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (parseDate(dateFrom).isAfter(parseDate(dateTo))) {
+            return null;
+        }
+        StringBuilder resultText = new StringBuilder();
+        resultText.append("Отчёт за период ").append(dateFrom)
+                .append(" - ").append(dateTo).append("\n");
+        int salaryCounter = 0;
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                if (data[j].contains(names[i]) && checkDateRange(data[j], dateFrom, dateTo)) {
+                    int workedHours = Integer.parseInt(data[j].split(" ")[2]);
+                    int salaryPerHour = Integer.parseInt(data[j].split(" ")[3]);
+                    salaryCounter += workedHours * salaryPerHour;
+                }
+            }
+            resultText.append(names[i]).append(" - ").append(salaryCounter).append("\n");
+            salaryCounter = 0;
+        }
+        return resultText.toString();
+    }
+
+    private static LocalDate parseDate(String date) {
```

Comment:
@kmety you do not need this method. Check the available methods in LocalDate class